imap_processing/tests/ultra/unit/test_ultra_l1b_extended.py

```diff
@@ -85,3 +87,35 @@ def test_get_ph_tof_and_back_positions(
 
     np.testing.assert_array_equal(ph_xb, selected_rows["Xb"].astype("float"))
     np.testing.assert_array_equal(ph_yb, selected_rows["Yb"].astype("float"))
+
+
+def test_get_ssd_back_position_and_tof_offset(
+    de_dataset,
+    events_fsw_comparison_theta_0,
+):
+    """Tests get_ssd_back_position function."""
+    yb, tof_offset, ssd_number = get_ssd_back_position_and_tof_offset(de_dataset)
+
+    df = pd.read_csv(events_fsw_comparison_theta_0)
+    df_filt = df[(df["StartType"] != -1) & (df["StopType"] >= 8)]
+
+    np.testing.assert_array_equal(yb, df_filt["Yb"].astype("float"))
+
+    tof_offset_lt = tof_offset[df_filt["StartType"] == StartType.Left.value]
+    tof_offset_rt = tof_offset[df_filt["StartType"] == StartType.Right.value]
+
+    ssd_number_lt = ssd_number[df_filt["StartType"] == StartType.Left.value]
+    ssd_number_rt = ssd_number[df_filt["StartType"] == StartType.Right.value]
+
+    np.testing.assert_array_equal(
+        tof_offset_lt[ssd_number_lt == 3],
+        np.full(len(tof_offset_lt[ssd_number_lt == 3]), -4.2),
```

Comment:
Is it possible that there could be other ssd numbers in tof_offset_lt and tof_offset_rt? Since you're filtering it for a specific value and using that in the assertion, but you're testing the entire object returned from `get_ssd_back_position_and_tof_offset`, it might be good to add an assertion that there are not other ssd numbers in the tof_offset besides the ones you're filtering on.

Let me know if it sounds like I'm misunderstanding your test.

Reply:
Good idea to add another test. There should always be 8 SSDs.